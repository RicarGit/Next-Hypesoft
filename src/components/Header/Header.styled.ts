import styled, { css } from "styled-components"

export const Header = styled.header`
  height: 1080px;
  width: 100%;
`

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: inherit;
  width: 1320px;
  margin: 0 auto;
`

export const ImageContainer = styled.div`
  position: absolute;
  height: 1080px;
  width: 1053px;
  padding-top: 106px;
  top: 0;
  right: 0;
  z-index: -1;
`

export const ImageBg = styled.div`
  position: absolute;
  height: inherit;
  width: 724px;
  right: 0;
  top: 0;
  background-color: ${({ theme }) => theme.colors.sec};
`

export const DotImageWrapper = styled.div`
  position: absolute;
  top: 417px;
  right: 325px;
`

export const HeroContainer = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  width: 758px;
  height: 412px;
  top: 334px;
`

export const HeroTitle = styled.h1`${({ theme }) => css`
  position: relative;
  color: ${theme.colors.white};
  font-family: ${theme.font.family};
  font-size: 80px;
  font-weight: 400;
  line-height: 98px;

  & span {
    color: ${theme.colors.sec}
  }

  &::after {
    content: '';
    display: block;
    position: absolute;
    bottom: 8px;
    left: 15px;
    height: 36px;
    width: 594px;
    background-color: ${({ theme }) => theme.colors.sec};
    opacity: 0.15;
  }
`}
`

export const HeroDesc = styled.p`
  width: 673px;
  margin-top: 32px;
  font-size: 20px;
  font-weight: 400;
  line-height: 32px;
  color: ${({ theme }) => theme.colors.gray[500]};
`

export const HeroButtonsWrapper = styled.div`
  display: flex;
  margin-top: 32px;
  gap: 16px;
`

export const HeroButton = styled.button`
  padding: 0 32px;
  font-size: 16px;
  line-height: 56px;
  font-weight: 600;
  letter-spacing: 0.02em;
  text-transform: uppercase;
  border: none;
  border-radius: 5px;
  background-color: ${({ theme }) => theme.colors.white};
  cursor: pointer;

  &.active {
    color: ${({ theme }) => theme.colors.white};
    background-color: ${({ theme }) => theme.colors.sec};
  }
`

export const FollowMeOn = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 228px;
  height: 84px;
  top: 948px;
  color: ${({ theme }) => theme.colors.white};
  line-height: 24px;
`

export const FollowMeButtonsContainer = styled.div`
  display: flex;
  gap: 12px;
`