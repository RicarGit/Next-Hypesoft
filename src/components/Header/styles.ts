import styled, { css } from "styled-components"

export const Header = styled.header`
  height: 100%;
  width: 100%;
`

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 1320px;
  margin: 0 auto;
`

export const ImageContainer = styled.div`
  position: absolute;
  height: 1080px;
  width: 1053px;
  padding-top: 106px;
  top: 0;
  right: 0;
  z-index: -1;
`

export const ImageBg = styled.div`
  position: absolute;
  height: inherit;
  width: 724px;
  right: 0;
  top: 0;
  background-color: ${({ theme }) => theme.colors.sec};
`

export const DotImageWrapper = styled.div`
  position: absolute;
  top: 417px;
  right: 325px;
`

export const HeroContainer = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  width: 758px;
  height: 412px;
  top: 334px;
`

export const HeroTitle = styled.h1`${({ theme }) => css`
  position: relative;
  color: ${theme.colors.white};
  font-family: ${theme.font.family};
  font-size: 80px;
  font-weight: 400;
  line-height: 98px;

  & span {
    color: ${theme.colors.sec}
  }

  &::after {
    content: '';
    display: block;
    position: absolute;
    bottom: 8px;
    left: 15px;
    height: 36px;
    width: 594px;
    background-color: ${({ theme }) => theme.colors.sec};
    opacity: 0.15;
  }
`}
`

export const HeroDesc = styled.p`
  width: 673px;
  margin-top: 32px;
  font-size: 20px;
  font-weight: 400;
  line-height: 32px;
  color: ${({ theme }) => theme.colors.gray[500]};
`

export const HeroButtonsWrapper = styled.div`
  display: flex;
  margin-top: 32px;
  gap: 16px;
`

type ButtonColors = {
  buttonColors: {
    color: string
    bgColor: string
  }
}

export const HeroButton = styled.button<ButtonColors>`
  ${({ theme, buttonColors: { color, bgColor } }) => css`
    padding: 0 32px;
    font-size: 16px;
    line-height: 56px;
    font-weight: 600;
    letter-spacing: 0.02em;
    text-transform: uppercase;
    border: none;
    border-radius: 5px;
    color: ${color};
    background-color: ${bgColor === 'white' ? theme.colors.white : theme.colors.sec};
    cursor: pointer;
  `}
`

export const FollowMeOn = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 228px;
  height: 84px;
  top: 948px;
  color: ${({ theme }) => theme.colors.white};
  line-height: 24px;
`

export const FollowMeButtonWrapper = styled.div`
  display: flex;
  gap: 12px;
`

type BgFollowMeButton = {
  bgColor: string
}

export const FollowMeButton = styled.button<BgFollowMeButton>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 14px;
  border: none;
  border-radius: 3px;
  background-color: ${({ theme, bgColor }) => bgColor === 'white'
    ? theme.colors.white
    : theme.colors.sec
  };
`