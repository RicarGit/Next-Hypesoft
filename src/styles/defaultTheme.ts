import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme