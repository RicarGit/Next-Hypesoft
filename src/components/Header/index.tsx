import * as S from './styles'
import Navbar from '../Navbar'

import Image from 'next/image'
import headerImage from './header-Image.svg'
import facebook from './facebook.svg'
import behance from './behance.svg'
import dribble from './dribbble.svg'
import twitter from './twitter.svg'
import darkVerticalDots from './dark-vertical-dots.svg'

const Header = () => {
  return (
    <S.Header>

      <Navbar />
      <S.Container>

        <S.ImageContainer>

          <S.ImageBg />
          <Image src={headerImage} />

          <S.DotImageWrapper>
            <Image src={darkVerticalDots} className={'darkVerticalDots'} />
          </S.DotImageWrapper>

        </S.ImageContainer>

        <S.HeroContainer>

          <S.HeroTitle>
            Hello, I’m John, a <span>Software Engineer.</span>
          </S.HeroTitle>

          <S.HeroDesc>
            Product Designer, UI/UX Designer, and developer based in Brazil.
            Over the past 17 years, as an art director and designer,
            I’ve worked with big companies and up-and-coming startups.
          </S.HeroDesc>

          <S.HeroButtonsWrapper>

            <S.HeroButton buttonColors={{ color: 'white', bgColor: 'sec' }}>
              Hire me
            </S.HeroButton>
            <S.HeroButton buttonColors={{ color: 'primary', bgColor: 'white' }}>
              View work
            </S.HeroButton>

          </S.HeroButtonsWrapper>

        </S.HeroContainer>

        <S.FollowMeOn>
          Follow me on
          <S.FollowMeButtonWrapper>

            <S.FollowMeButton bgColor={'white'}>
              <Image src={dribble} />
            </S.FollowMeButton>

            <S.FollowMeButton bgColor={'sec'}>
              <Image src={behance} />
            </S.FollowMeButton>

            <S.FollowMeButton bgColor={'white'}>
              <Image src={facebook} />
            </S.FollowMeButton>

            <S.FollowMeButton bgColor={'white'}>
              <Image src={twitter} />
            </S.FollowMeButton>

          </S.FollowMeButtonWrapper>

        </S.FollowMeOn>

      </S.Container>

    </S.Header>
  )
}

export default Header