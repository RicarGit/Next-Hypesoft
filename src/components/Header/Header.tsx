import * as S from './Header.styles'

import Image from 'next/image'
import { Assets } from '~/assets'

import { Navbar } from '~/components/Navbar'
import { HeaderSocialMediaButton } from '~/components/HeaderSocialMediaButton'

export const Header = () => {
  return (
    <S.Header>
      <Navbar />

      <S.Container>
        <S.ImageContainer>
          <S.ImageBg />

          <Image
            src={Assets.PersonalPhoto}
            width={870}
            height={974}
            alt='personal photo'
            priority
          />
          <S.DotImageWrapper>
            <Image
              src={Assets.DarkVerticalDots}
              width={68}
              height={152}
              alt='dark vertical dots'
            />
          </S.DotImageWrapper>
        </S.ImageContainer>

        <S.HeroContainer>
          <S.HeroTitle>
            Hello, I’m John, a <span>Software Engineer.</span>
          </S.HeroTitle>

          <S.HeroDesc>
            Product Designer, UI/UX Designer, and developer based in Brazil.
            Over the past 17 years, as an art director and designer,
            I’ve worked with big companies and up-and-coming startups.
          </S.HeroDesc>

          <S.HeroButtonsWrapper>
            <S.HeroButton className='active'>
              Hire me
            </S.HeroButton>

            <S.HeroButton>
              View work
            </S.HeroButton>
          </S.HeroButtonsWrapper>
        </S.HeroContainer>

        <S.FollowMeOn>
          Follow me on
          <S.FollowMeButtonsContainer>
            <HeaderSocialMediaButton
              icon={Assets.Dribble}
              alt='dribble logo'
            />
            <HeaderSocialMediaButton
              icon={Assets.Behance}
              alt='behance logo'
              className='active'
            />
            <HeaderSocialMediaButton
              icon={Assets.Facebook}
              alt='facebook logo'
            />
            <HeaderSocialMediaButton
              icon={Assets.Twitter}
              alt='twitter logo'
            />
          </S.FollowMeButtonsContainer>
        </S.FollowMeOn>
      </S.Container>
    </S.Header>
  )
}