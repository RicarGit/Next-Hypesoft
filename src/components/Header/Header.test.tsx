import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import Header from '.'

describe('Header', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Header />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})