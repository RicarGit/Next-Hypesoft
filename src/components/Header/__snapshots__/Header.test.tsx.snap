// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="sc-bczRLJ jiBWDJ"
>
  <nav
    class="sc-iqcoie QEpVZ"
  >
    <a
      class="sc-crXcEl jdvQZA"
      href="/#about"
    >
      about
    </a>
    <a
      class="sc-crXcEl jdvQZA"
      href="/#services"
    >
      services
    </a>
    <a
      class="sc-crXcEl jdvQZA"
      href="/#works"
    >
      works
    </a>
    <a
      class="sc-crXcEl jdvQZA"
      href="/#contact"
    >
      contact
    </a>
  </nav>
  <div
    class="sc-gsnTZi bmcOzO"
  >
    <div
      class="sc-dkzDqf gsHUQR"
    >
      <div
        class="sc-hKMtZM enyiLn"
      />
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="personal image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="sc-eCYdqJ iiZQb"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="dark vertical dots"
            class="darkVerticalDots"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="sc-jSMfEi hfTTNQ"
    >
      <h1
        class="sc-gKXOVf eSuXcz"
      >
        Hello, I’m John, a 
        <span>
          Software Engineer.
        </span>
      </h1>
      <p
        class="sc-iBkjds kyYlfY"
      >
        Product Designer, UI/UX Designer, and developer based in Brazil. Over the past 17 years, as an art director and designer, I’ve worked with big companies and up-and-coming startups.
      </p>
      <div
        class="sc-ftvSup duEtcv"
      >
        <button
          class="sc-papXJ gHKZvi active"
        >
          Hire me
        </button>
        <button
          class="sc-papXJ gHKZvi"
        >
          View work
        </button>
      </div>
    </div>
    <div
      class="sc-jqUVSM hgQuJV"
    >
      Follow me on
      <div
        class="sc-kDDrLX kAmMJW"
      >
        <button
          class="sc-evZas kuVVQE"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="social media icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
        <button
          class="sc-evZas kuVVQE active"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="social media icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
        <button
          class="sc-evZas kuVVQE"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="social media icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
        <button
          class="sc-evZas kuVVQE"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="social media icon"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
      </div>
    </div>
  </div>
</header>
`;
